xlators/protocol/client/src/client-helpers.c

```diff
@@ -417,6 +418,10 @@ client_get_remote_fd(xlator_t *this, fd_t *fd, int flags, int64_t *remote_fd)
             if (fd->anonymous) {
                 *remote_fd = GF_ANON_FD_NO;
             } else {
+                if (conf->strict_locks &&
+                    (fop == GFS3_OP_WRITE || fop == GFS3_OP_FTRUNCATE)) {
+                    locks_involved = _gf_true;
+                }
```

Comment:
Why only these two, what about fallocate/discard/zerofill?

Reply:
Added them as well.

Reply:
Could you tell me the how one decides which fops to disallow?

Reply:
Currently we are disallowing data transactions which can lead to data inconsistency between the replicas.